.github/workflows/release.yml

```diff
@@ -0,0 +1,57 @@
+name: release
+on:
+  push:
+    tags:
+      - '*'
+  workflow_dispatch:
+
+jobs:
+
+  build-sdist:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout
+        uses: actions/checkout@v2
+      - run: git fetch --prune --unshallow
+
+      - name: Install Python
+        uses: actions/setup-python@v2
+        with:
+          python-version: 3.6
+
+      - name: Setup apt
+        run: |
+          sudo apt update
+          sudo apt install -y libeigen3-dev
+
+      - name: Setup
+        run: |
+          python3 -m pip install --upgrade pip
+          pip3 install build
+
+      - name: Build sdist
+        run: python3 -m build --sdist -o dist/
+
+      - name: Build wheel
+        run: python3 -m build --wheel -o dist/
+
+      - name: Upload artifacts
+        uses: actions/upload-artifact@v2
+        with:
+          path: dist/*
+          name: dist
+
+  upload_pypi:
+    needs: build-sdist
+    runs-on: ubuntu-latest
+    steps:
+
+      - uses: actions/download-artifact@v2
+        with:
+          name: artifact
+          path: dist
+
+      - uses: pypa/gh-action-pypi-publish@master
+        with:
+          user: __token__
+          password: ${{ secrets.PYPI_TOKEN }}
```

Comment:
I just realized that this will fail. By default, wheels with C/C++ extensions cannot be uploaded to PyPI, they need to be processed by tools like [pypa/auditwheel](https://github.com/pypa/auditwheel) to make them compatible with `manylinux*`. Making this work in a multi-platform setup is quite challenging, and I'd rather leave uploading wheels to a now PR if you don't mind, it might take few iterations to make it work.

However, this workflow could remain and just upload the source distribution, what do you think? @artivis

Reply:
Exactly my plan. Revisit this workflow in a separate PR before the next tag. I'll comment this specific job so that it is more explicit.

Reply:
Just the workflow and commented this step as per your comments.